### PR TITLE
bpo-44539: Support recognizing JPEG files without JFIF or Exif markers

### PR DESCRIPTION
**Pull request by mohamadmansourX**
Previous method to check JPG images was using the following command `h[6:10] in (b'JFIF', b'Exif')`
However, its not always the case as some might start with `b'\xff\xd8\xff\xdb'` header.
`\xdb` defining the Quantization Table.

Reference:
https://www.digicamsoft.com/itu/itu-t81-36.html
https://web.archive.org/web/20120403212223/http://class.ee.iastate.edu/ee528/Reading%20material/JPEG_File_Format.pdf

As an example, the attached image returned None from the `imghdr.what(filename)`

https://bugs.python.org/issue44539

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).



## CLA Missing

Our records indicate the following people have not signed the CLA:

@mohamadmansourX

For legal reasons we need all the people listed to sign the CLA before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.

If you have recently signed the CLA, please wait at least one business day
before our records are updated.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

**Comment by fbidu** [comment]
Thanks for the PR, @mohamadmansourX. I was able to test this PR on a `Linux Mint 20.1 x86_64` machine.

0. The supplied sample image when analyzed through the `file` command returns `JPEG image data, baseline, precision 8, 500x332, components 3`
1. Testing the current code for `imghdr.what` with the supplied sample image returns `None`.
2. After applying the patch, `imghdr.what` positively ids the image, returning `jpeg`

It would be nice to have a test case for this over `Lib/test/test_imghdr.py`. I see there are some tests there that involve only passing a hardcoded header and others that load images from `Lib/test/imghdrdata`. I think that at least a test with the harcoded header is welcome.

Also, please see the message about the CLA. There's also something wrong with the news entry but I couldn't figure that out

File: Lib/imghdr.py [hunk 1]
```diff
@@ -38,6 +38,8 @@ def test_jpeg(h, f):
     """JPEG data in JFIF or Exif format"""
     if h[6:10] in (b'JFIF', b'Exif'):
         return 'jpeg'
+    elif h[:4]==b'\xff\xd8\xff\xdb':
```
Comment (merwok) [inline Lib/imghdr.py]: For better readability:

```suggestion
    elif h[:4] == b'\xff\xd8\xff\xdb':
```

File: Misc/NEWS.d/next/Library/2021-06-30-11-34-35.bpo-44539.nP0Xi4.rst [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+#.. section: macOS
+#.. section: IDLE
+#.. section: Tools/Demos
+#.. section: C API
+
+# Write your Misc/NEWS entry below.  It should be a simple ReST paragraph.
+# Don't start with "- Issue #<n>: " or "- bpo-<n>: " or that sort of stuff.
+###########################################################################
+
+
```
Comment (merwok) [inline Misc/NEWS.d/next/Library/2021-06-30-11-34-35.bpo-44539.nP0Xi4.rst]: Could you remove all the commented lines?

Comment (fbidu) [reply Misc/NEWS.d/next/Library/2021-06-30-11-34-35.bpo-44539.nP0Xi4.rst]: @merwok do you have any clues regarding the doc error? I tried removing the commented-out lines and running the sphinx toolchain locally but it all broke in the same way :/

**Comment by merwok** [comment]
It seems that some issue with the blurb filename hits a bug in blurb:

https://app.travis-ci.com/github/python/cpython/jobs/523615771

```
 File "/home/travis/virtualenv/python3.6.10/lib/python3.6/site-packages/blurb.py", line 486, in throw
    raise BlurbError(f("Error in {filename}:{line_number}:\n{s}"))
  File "/home/travis/virtualenv/python3.6.10/lib/python3.6/site-packages/blurb.py", line 138, in f
    return s.format_map(d)
KeyError: 'filename'
```

**Comment by merwok** [comment]
It’s known and fixed: https://github.com/python/core-workflow/pull/386

Maybe blurb-it needs an update.

**Pull request merged**